src/AtmClient.php

```diff
@@ -176,4 +188,146 @@ public function createLockedContent($content_id, $property_id, $text) {
     return FALSE;
   }
 
+  /**
+   * Create new ATM Property.
+   *
+   * @return array|bool
+   *   An array of property config options or false on error.
+   */
+  public function createAtmProperty() {
+    $client = new Client($this->getConfig());
+
+    try {
+      $request = $client->put($this->atmHost . '/atm-admin/property/create', [
+        'headers' => $this->getConfig('headers'),
+        'json' => [
+          'Name' => 'Drupal:' . \Drupal::config('system.site')->get('name'),
+          'Website' => \Drupal::request()->getHost(),
+        ],
+      ]);
+
+      $response = Json::decode($request->getBody()->getContents());
+
+      if ($this->atmConfig->get('development')) {
+        \Drupal::logger('adtechmedia')
+          ->debug('ATM Property ID: @property', [
+            '@property' => $response['Id'],
+          ]);
+      }
+
+      return $response;
+    }
+    catch (RequestException $e) {
+      watchdog_exception('adtechmedia', $e->getMessage());
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Retrieve ATM Property.
+   *
+   * @return array|bool
+   *   An array of property config options or false on error.
+   */
+  public function retrieveAtmProperty() {
+    $client = new Client($this->getConfig());
+    $this->setQuery('Id', $this->propertyId);
+
+    try {
+      $request = $client
+        ->get($this->atmHost . '/atm-admin/property/retrieve', $this->options);
+      $response = Json::decode($request->getBody()->getContents());
+
+      return $response;
+    }
+    catch (RequestException $e) {
+      watchdog_exception('adtechmedia', $e->getMessage());
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Update ATM Property.
+   *
+   * @param array $config
+   *   ATM properties.
+   *
+   * @return bool|mixed
+   *   Updated property.
+   */
+  public function updateAtmProperty($config) {
+    $client = new Client($this->getConfig());
+
+    try {
+      $request = $client->patch($this->atmHost . '/atm-admin/property/update-config', [
+        'headers' => $this->getConfig('headers'),
+        'json' => [
+          'Id' => $this->propertyId,
+          'ConfigDefaults' => [
+            'revenueMethod' => $config['revenue_model'],
+            'content' => [
+              'offsetType' => $config['content_preview_type'],
+              'offset' => $config['content_preview'],
+              'lock' => $config['locking_algorithm'],
+            ],
+            'styles' => ['main' => $config['styles']],
+            'ads' => [
+              'relatedVideoCb' => "function(onReady) { onReady({ 'iPad,iPhone,Apple': ['https://www.youtube.com/watch?v=OzQh6wDb2oE&t=23s'] }) }",
```

Comment:
The `onReady` call should look like this: `onReady('https://www.youtube.com/watch?v=OzQh6wDb2oE&t=23s')`